### PR TITLE
NIFI-13619 Switch web-client from OkHttp to Java HttpClient

### PR DESCRIPTION
**Pull request by exceptionfactory**
# Summary

[NIFI-13619](https://issues.apache.org/jira/browse/NIFI-13619) Switches the implementation of `nifi-web-client` from [OkHttp](https://square.github.io/okhttp/) to the Java [HttpClient](https://docs.oracle.com/en/java/javase/21/docs/api/java.net.http/java/net/http/HttpClient.html) class, implementing the existing `WebClientService` interface without the additional dependencies, such as Kotlin standard libraries. The Java `HttpClient` supports standard HTTP/2 and HTTP/1.1 protocols along with proxy access and authentication.

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardHttpUriBuilder.java [hunk 1]
```diff
@@ -16,65 +16,171 @@
...
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 
 /**
  * Standard HTTP URI Builder based on OkHttp HttpUrl
```
Comment (joewitt) [inline nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardHttpUriBuilder.java:33]: no longer OkHttp HttpUrl

Comment (exceptionfactory) [reply nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardHttpUriBuilder.java]: Thanks @joewitt! I pushed an update to correct the comments.

File: nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardWebClientService.java [hunk 2]
```diff
@@ -87,17 +98,16 @@ public void setConnectTimeout(final Duration connectTimeout) {
      */
     public void setReadTimeout(final Duration readTimeout) {
         Objects.requireNonNull(readTimeout, "Read Timeout required");
-        okHttpClient = okHttpClient.newBuilder().readTimeout(readTimeout).build();
+        this.readTimeout = readTimeout;
     }
 
     /**
-     * Set timeout for writing requests to socket connection
+     * Set timeout for writing requests to socket connection is not supported with current HTTP Client implementation
```
Comment (joewitt) [inline nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardWebClientService.java:105]: It does seem possible to set a timeout on requests

https://www.baeldung.com/java-httpclient-timeout

I dont quite follow in the new api read/write timeout but it does seem like you can set a timeout on the requests.

Comment (exceptionfactory) [reply nifi-commons/nifi-web-client/src/main/java/org/apache/nifi/web/client/StandardWebClientService.java]: Thanks @joewitt. The changes already included handling of the `readTimeout` property, passing it to the `HttpRequest.Builder`. However, for the sake of completeness, I added handling for `writeTimeout`, but noted that `readTimeout` is preferred. OkHttp had separate methods with different timeout behavior, but the latest commit optionally uses `writeTimeout` for the request.

**Review by bbende — APPROVED** [review]
+1 Code looks good, ran this change with some existing code that leverages the `WebClientService` with `TLS` and `non-TLS` and everything still worked, will merge

**Pull request merged**